dashboard/app/controllers/lti_v1_controller.rb

```diff
@@ -407,6 +408,96 @@ def new_integration
+  def new_registration
+    begin
+      params.require([:openid_configuration, :registration_token])
+      openid_configuration_url = params[:openid_configuration]
+      registration_token = params[:registration_token]
+      openid_config = JSON.parse(HTTParty.get(openid_configuration_url).body)
+    rescue => exception
+      message = 'Error getting LMS openid_configuration'
+      Honeybadger.notify(exception, context: {message: message})
+      return render status: :internal_server_error, json: {error: message}
+    end
+    # Create registration ID to pass to be used by the view as a hidden form
+    # field. This ID will be useed to read the registration_data back out from
+    # the cache in the create_registration method
+    @registration_id = SecureRandom.uuid
+    registration_data = {
+      registration_token: registration_token,
+      registration_endpoint: openid_config['registration_endpoint'],
+      issuer: openid_config['issuer'],
+      lms_account_name: openid_config.dig(Policies::Lti::LTI_PLATFORM_CONFIGURATION, Policies::Lti::CANVAS_ACCOUNT_NAME),
+    }
+    # Expire cache in 1 hour to match expiration of registration token
+    write_cache(@registration_id, registration_data, 1.hour)
+    render 'lti/v1/dynamic_registration', layout: false
+  end
+
+  # POST /lti/v1/dynamic_registration
+  # Initiate LTI Dynamic Registration Request and create LTI Integration
+  def create_registration
+    begin
+      params.require([:email])
+    rescue ActionController::ParameterMissing => exception
+      return render json: {error: exception.message}, status: :bad_request
+    end
+
+    admin_email = params[:email]
+    cache_key = params[:registration_id]
+    # Get cached values
+    registration_data = read_cache(cache_key)
+
+    # Don't proceed if expected registration data doesn't exist
+    return unauthorized_status if registration_data.nil?
+
+    platform = Policies::Lti.find_platform_by_issuer(registration_data[:issuer])
+    if platform.nil?
+      message = "Unsupported issuer: #{registration_data[:issuer]}"
+      Honeybadger.notify(
+        'LTI dynamic registration error',
+        context: {
+          message: message,
+        }
+      )
+      return render status: :unprocessable_entity, json: {error: message}
+    end
+    # Create registration request data
+    body = Policies::Lti::DYNAMIC_REGISTRATION_CONFIG.to_json
+    headers = {
+      'Content-Type' => 'application/json',
+      'Authorization' => "Bearer #{registration_data[:registration_token]}"
+    }
+
+    begin
+      response = HTTParty.post(registration_data[:registration_endpoint], body: body, headers: headers)
+      registration_response = JSON.parse(response.body).symbolize_keys
+    rescue => exception
+      message = 'Error creating registration'
+      Honeybadger.notify(exception, context: {message: message})
+      return render status: :internal_server_error, json: {error: message}
+    end
```

Comment:
Would it be worth pulling this section into its own function or class (maybe a service or client)? Would keep the size of this route down, and in tests you could just stub the new function instead of having to stub a specific http request.